hedera-node/docs/design/services/smart-contract-service/evm-transaction-response-codes.md

```diff
@@ -0,0 +1,120 @@
+
+- **MAX_CONTRACT_STORAGE_EXCEEDED**
+  - Contract permanent storage exceeded the currently allowable limit.
+
+- **MAX_STORAGE_IN_PRICE_REGIME_HAS_BEEN_USED**
+  - All contract storage allocated to the current price regime has been consumed.
+
+- **MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED**
+  - The maximum number of entities allowed in the current price regime have been created.
+
+## Additional statuses, that are not propagated from child transactions, but can be returned as top-level status:
+Those status codes can also be present for `ContractCall`, `ContractCreate` via `EthereumTransaction`.
+
+- **CONTRACT_NEGATIVE_VALUE**
+  - Negative value/initial balance was specified in a smart contract call/create.
+
+- **CONTRACT_DELETED**
+  - Contract is marked as deleted.
+  - Scope: This can happen as a top-level status only for contract call and ethereum contract call.
+
+## Create operation Specific Response Codes
+These responses do include ContractCreate via EthereumTransaction.
+
+- **AUTORENEW_DURATION_NOT_IN_RANGE**
+  - The duration is not a subset of `[MINIMUM_AUTORENEW_DURATION, MAXIMUM_AUTORENEW_DURATION]`.
+
+- **CONTRACT_NEGATIVE_GAS**
+  - Negative gas was offered in a smart contract call.
+
+- **MAX_GAS_LIMIT_EXCEEDED**
+  - Gas exceeded the currently allowable gas limit per transaction.
+
+- **INVALID_MAX_AUTO_ASSOCIATIONS**
+  - The maximum automatic associations value is not valid. The most common cause for this error is a value less than `-1`.
+
+- **INVALID_AUTORENEW_ACCOUNT**
+  - The `autoRenewAccount` specified is not a valid, active account.
+
+- **ERROR_DECODING_BYTESTRING**
+  - Decoding the smart contract binary to a byte array failed. Check that the input is a valid hex string.
+
+- **INVALID_FILE_ID**
+  - The file ID is invalid or does not exist.
+
+- **FILE_DELETED**
+  - The file has been marked as deleted.
+
+- **CONTRACT_BYTECODE_EMPTY**
+  - Bytecode for the smart contract is of length zero.
+
+- **CONTRACT_FILE_EMPTY**
+  - The file to create a smart contract was of length zero.
+
+- **INVALID_ACCOUNT_ID**
+  - The account ID is invalid or does not exist.
+
+## Ethereum Transaction Specific Response Codes
+These are additional response codes that can appear specifically for `EthereumTransaction`.
+
+- **WRONG_NONCE**
+  - This transaction specified an `ethereumNonce` that is not the current `ethereumNonce` of the account.
+
+- **WRONG_CHAIN_ID**
+  - `EthereumTransaction` was signed against a chain ID that this network does not support.
+
+- **INVALID_ETHEREUM_TRANSACTION**
+  - The Ethereum transaction either failed parsing or failed signature validation, or some other `EthereumTransaction` error not covered by another response code.
+
+Note: The description for the statuses is copied from [protobuf repo](https://github.com/hashgraph/hedera-protobufs/blob/main/services/response_code.proto)
```

Comment:
I would move this line up in the file - near the top really.  (Say, after L10)

Reply:
Done :)